rust/lance/src/dataset/scanner.rs

```diff
@@ -883,6 +925,67 @@ impl Scanner {
             FilterPlan::default()
         };
 
+        if let Some(query) = &self.full_text_query {
+            let index_info = self.dataset.scalar_index_info().await?;
+            let columns = if query.columns.is_empty() {
```

Comment:
The `create_plan` method is already quite complicated.  Can we put all the logic in this if branch into a separate method?  It should only need `&mut filter_plan`